### PR TITLE
Submit oscarshiang's project proposal

### PR DESCRIPTION
**Pull request by OscarShiang**
Create `oscarshiang/README.rst` to describe the details and the schedule of the project.

---
A good PR should include the following parts:

- [x] Provide a clear title
- [x] Provide meaningful descriptions in the PR and the commit logs
- [x] Name your branch like ``<github-user-name>-hw#``, e.g., ``myname-hw2-attempt1``
- [x] Merge from your specific branch (***not master***) to the `hw*` branch (***not master***, again)
- [x] Exclude irrelevant files, such as binaries, text files, or dot files
- [x] Make sure tests/CI pass

### CODE REVIEW

File: project/OscarShiang/README.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+======
+PicoLA
```
Comment (yungyuc) [inline project/OscarShiang/README.rst:2]: I like the name.

File: project/OscarShiang/README.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+======
+
+PicoLA: a compact C++ library for doing linear algebra computations
+
+Basic Information
+=================
+
+`PicoLA` aims to provide coding experience similiar to MATLAB and is capable to run on resource-limited devices without installing extra dependent packages.
```
Comment (yungyuc) [inline project/OscarShiang/README.rst]: Very clear goal.

File: project/OscarShiang/README.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
+======
+PicoLA
+======
+
+PicoLA: a compact C++ library for doing linear algebra computations
```
Comment (yungyuc) [inline project/OscarShiang/README.rst:5]: Please supply the github repository.

File: project/OscarShiang/README.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+Linear algebra is frequently used by real-world applications. It does not only appear in large scale high performance servers, it usually needs to be shipped with embedded devices as products as well.
+
+However, porting linear algebra stuffes from idea to product sometimes become a pain for the developers. Since the syntax and the properties of MATLAB script may differs from the languages used by embedded developers. Even if it works, it could be not easy to solve the dependencies of the library and impossible to utilize on bare-metal platforms.
+
+The library tries to deal with the problem by supporting similiar syntax using by MATLAB developers and not to compromise the performance in the same time.
+
+The following operations are considered to be implemented:
+
+1. Vector and matrix generation
+2. Related operations (e.g. add, substract, multiply, cross, normalize, sum, average, ...)
```
Comment (yungyuc) [inline project/OscarShiang/README.rst]: Please write down a complete list of operations to support in your project.

File: project/OscarShiang/README.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+- Automatic build and test system: GitHub Action
+
+Since the whole project is composed by C++, we choose GoogleTest as the unit-testing framework. Leverage GitHub Actions to do testing whenever the code is pushed onto branches. As GitHub Action supports Apple M1 Machines, we can ensure that all the hardware-related instructions can be properly compiled and run on actual machines.
+
+For public APIs, we try use [hdoc](https://hdoc.io/) to generate human-readable document. This framework can also be builded and released via GitHub Action. So that we can simply host the document page on GitHub.
+
+Schedule
+========
+
+- Week 1: Setup the infrastructures
```
Comment (yungyuc) [inline project/OscarShiang/README.rst]: Please add dates in the schedule.

File: project/OscarShiang/README.rst [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+- Week 2: Define the library's data structures
+- Week 3: Implement fundamental operations for vector and matrix
+- Week 4: Research for implement vector and matrix operations with SIMD instructions
+- Week 5: Add SIMD support into the library
+- Week 6: Create unit tests and set onto GitHub Actions
```
Comment (yungyuc) [inline project/OscarShiang/README.rst]: Unit testing and CI are to improve your productivity.  Doing them in the final phase of your development defeats the purpose.  Please revise the plan to start them in the beginning.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [x] Add an entry in `project/README.rst` pointing to your proposal.
- [x] Change to use less than 80 characters in a line.  Long lines are hard to review and annotate.
- [x] [L5](https://github.com/yungyuc/nsdhw_24au/pull/31/files#diff-7ae2f9e3546651a97bd5b524fd37bdbeb8860b16a1d06642253c887a929f7d90R5): Github repository.
- [x] [L24](https://github.com/yungyuc/nsdhw_24au/pull/31/files#diff-7ae2f9e3546651a97bd5b524fd37bdbeb8860b16a1d06642253c887a929f7d90R24): Explicit list of operations.
- [x] [L68](https://github.com/yungyuc/nsdhw_24au/pull/31/files#diff-7ae2f9e3546651a97bd5b524fd37bdbeb8860b16a1d06642253c887a929f7d90R68): Pull forward testing and CI.

**Comment by OscarShiang** [comment]
I have updated README.rst with the following changes:
- Add repo link in basic information section
- Specify all the operations need to be implemented
- Revise and add expected date into the schedule
- Translate the link syntax from markdown into rst
- Break all line that is has more than 80 characters

**Comment by OscarShiang** [comment]
The entry of my project has also been added into `project/README.rst`.

**Review by yungyuc — APPROVED** [review]
> I have updated README.rst with the following changes:
> 
> * Add repo link in basic information section
> * Specify all the operations need to be implemented
> * Revise and add expected date into the schedule
> * Translate the link syntax from markdown into rst
> * Break all line that is has more than 80 characters
>
> The entry of my project has also been added into `project/README.rst`.

LGTM.  Let me know if you want the PR to be merged.

**Comment by OscarShiang** [comment]
I am ready to be merged!

**Pull request merged**